casper/src/main/rholang/BasicWallet.rho

```diff
@@ -0,0 +1,67 @@
+//scalapackage coop.rchain.rholang.wallet
+
+//assumes standard purse API from MakeMint
+contract @"BasicWallet"(@purse, @algorithm, @pk, return) = {
+  new algorithmKnown, cryptoVerifyCh in {
+    match algorithm {
+      "ed25519" => {
+        algorithmKnown!(true) | cryptoVerifyCh!("ed25519Verify")
+      }
+      _ => {
+        algorithmKnown!(false)
+      }
+    } |
+    for(@ak <- algorithmKnown) {
+      match ak {
+        false => { return!([]) }
+        true => {
+          for (cryptoVerify <- cryptoVerifyCh) {
+            new basicWallet, verify, nonceCh in {
+              nonceCh!(-1) |
+              contract verify(@nonce, @sig, return) = {
+                for(@prevNonce <- nonceCh) {
+                  match nonce == (prevNonce + 1) {
```

Comment:
This design makes it easy to mount a denial-of-service attack: just request a nonce and never use it.